---
title: "Differential expression analysis of cauda EV small RNAs - all biotypes combined (including rRNAs) "
author: "Anar Alshanbayeva"
date: "5/26/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r , warning=FALSE, message=FALSE}
######################## LOAD LIBRARIES ########################
######################## LOAD LIBRARIES ########################
######################## LOAD LIBRARIES ########################

library(variancePartition)
library(synapser)
library(Biobase)
library(GEOquery)
library(GenomicRanges)
library(RcppRoll)
library(biomaRt)
library(data.table)
library(doParallel)
library(foreach)
library(genomation)
library(grid)
library(gridExtra)
library(knitr)
library(limma)
library(qvalue)
library(scales)

library('variancePartition')
library('edgeR')
library('BiocParallel')
library(biomaRt) 
library(dplyr)
library(tximport)
library(tximeta)
library(edgeR)
library(ggplot2)
library(msigdbr)
library(Rsubread) 
library(limma)
library(ggplot2)
library(DESeq2)
library(org.Mm.eg.db )
library(pheatmap)
library(edgeR)
library(reshape2)
library(plotly)
library(viridis)
library(cqn)
library(scales)
library(pheatmap)
library(gplots)
library(RColorBrewer) 
library (grDevices)
library(readr)
library(readxl)




### LOAD SAMPLES & COMBINE THE BIOTYPES COUNTS INTO 1 COUNT MATRIX from first sequencing batch ###
### LOAD SAMPLES & COMBINE THE BIOTYPES COUNTS INTO 1 COUNT MATRIX from first sequencing batch ###
### LOAD SAMPLES & COMBINE THE BIOTYPES COUNTS INTO 1 COUNT MATRIX  from first sequencing batch ###

### get the rRNA counts of all smaples & merge them to get 1 table with all counts (in the same order as all other biotype counts):###
### get the rRNA counts of all smaples & merge them to get 1 table with all counts (in the same order as all other biotype counts):###
### get the rRNA counts of all smaples & merge them to get 1 table with all counts (in the same order as all other biotype counts):###
rRNAs_2<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample2_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_3<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample3_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_4<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample4_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_5<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample5_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_6<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample6_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_7<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample7_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_8<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample8_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_9<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample9_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_11<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample11_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_14<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190308_sample14_rrna_counts.txt", header=TRUE, sep="", row.names=1)

merged_rrna_counts <- cbind(rRNAs_2, rRNAs_3,rRNAs_4,rRNAs_5,rRNAs_6,rRNAs_7,rRNAs_8,rRNAs_9,rRNAs_11,rRNAs_14)
colnames(merged_rrna_counts)<-c("MS28F1_sample2",  "MS28F1_sample3" , "MS28F1_sample4" , "MS28F1_sample5" , "MS28F1_sample6" , "MS28F1_sample7",  "MS28F1_sample8" , "MS28F1_sample9" , "MS28F1_sample11", "MS28F1_sample14")

tRNAs<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run3_ExceRpt_size_selected/caudaEVs_run3_miRNAsfirst_ExceRpt/exceRpt_tRNA_ReadCounts.txt", header=TRUE, sep="", row.names=1) # Loading the raw count matrices for each RNA biotype
piRNAs<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run3_ExceRpt_size_selected/caudaEVs_run3_miRNAsfirst_ExceRpt/exceRpt_piRNA_ReadCounts.txt", header=TRUE, sep="", row.names=1)
miRNAs<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run3_ExceRpt_size_selected/caudaEVs_run3_miRNAsfirst_ExceRpt/exceRpt_miRNA_ReadCounts.txt", header=TRUE, sep="", row.names=1)
gencode<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run3_ExceRpt_size_selected/caudaEVs_run3_miRNAsfirst_ExceRpt/exceRpt_gencode_ReadCounts.txt" , header=TRUE, sep="", row.names=1)
circularRNAs<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run3_ExceRpt_size_selected/caudaEVs_run3_miRNAsfirst_ExceRpt/exceRpt_circularRNA_ReadCounts.txt", header=TRUE, sep="", row.names=1)


tRNAs <- as.data.frame(tRNAs)
piRNAs<- as.data.frame(piRNAs)
miRNAs <- as.data.frame(miRNAs)
gencode <- as.data.frame(gencode)
circularRNAs <- as.data.frame(circularRNAs)
rRNAs <- as.data.frame(merged_rrna_counts)
counts_total_run3 <-rbind(tRNAs,piRNAs,miRNAs,gencode,circularRNAs,rRNAs)



### LOAD SAMPLES & COMBINE THE BIOTYPES COUNTS INTO 1 COUNT MATRIX from second sequencing batch ###
### LOAD SAMPLES & COMBINE THE BIOTYPES COUNTS INTO 1 COUNT MATRIX from second sequencing batch ###
### LOAD SAMPLES & COMBINE THE BIOTYPES COUNTS INTO 1 COUNT MATRIX from second sequencing batch ###

### get the rRNA counts of all smaples & merge them to get 1 table with all counts (in the same order as all other biotype counts):###
### get the rRNA counts of all smaples & merge them to get 1 table with all counts (in the same order as all other biotype counts):###
### get the rRNA counts of all smaples & merge them to get 1 table with all counts (in the same order as all other biotype counts):###
rRNAs_2<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample2_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_3<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample3_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_4<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample4_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_5<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample5_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_6<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample6_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_7<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample7_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_8<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample8_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_9<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample9_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_11<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample11_rrna_counts.txt", header=TRUE, sep="", row.names=1)
rRNAs_14<- read.table(file="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/counts/rRNA_counts/20190110_sample14_rrna_counts.txt", header=TRUE, sep="", row.names=1)

merged_rrna_counts <- cbind(rRNAs_2, rRNAs_3,rRNAs_4,rRNAs_5,rRNAs_6,rRNAs_7,rRNAs_8,rRNAs_9,rRNAs_11,rRNAs_14)
colnames(merged_rrna_counts)<-c("MS28F1_sample2",  "MS28F1_sample3" , "MS28F1_sample4" , "MS28F1_sample5" , "MS28F1_sample6" , "MS28F1_sample7",  "MS28F1_sample8" , "MS28F1_sample9" , "MS28F1_sample11", "MS28F1_sample14")

tRNAs<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run4_miRNAsfirst_ExceRpt/exceRpt_tRNA_ReadCounts.txt", header=TRUE, sep="", row.names=1) # Loading the raw count matrices for each RNA biotype
piRNAs<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run4_miRNAsfirst_ExceRpt/exceRpt_piRNA_ReadCounts.txt", header=TRUE, sep="", row.names=1)
miRNAs<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run4_miRNAsfirst_ExceRpt/exceRpt_miRNA_ReadCounts.txt", header=TRUE, sep="", row.names=1)
gencode<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run4_miRNAsfirst_ExceRpt/exceRpt_gencode_ReadCounts.txt" , header=TRUE, sep="", row.names=1)
circularRNAs<- read.table(file="/Users/alshanbayeva/data_analysis_september/caudaEVs_run4_miRNAsfirst_ExceRpt/exceRpt_circularRNA_ReadCounts.txt", header=TRUE, sep="", row.names=1)


tRNAs <- as.data.frame(tRNAs)
piRNAs<- as.data.frame(piRNAs)
miRNAs <- as.data.frame(miRNAs)
gencode <- as.data.frame(gencode)
circularRNAs <- as.data.frame(circularRNAs)
rRNAs <- as.data.frame(merged_rrna_counts)
counts_total_run4 <-rbind(tRNAs,piRNAs,miRNAs,gencode,circularRNAs,rRNAs)









### COMBINE THE MATRICES FROM 2 RUNS & ASSIGN FEATURES 1 RUN DOESN'T HAVE IN THE OTHER RUN AS "NA" VALUES  ###
### COMBINE THE MATRICES FROM 2 RUNS & ASSIGN FEATURES 1 RUN DOESN'T HAVE IN THE OTHER RUN AS "NA" VALUES  ###
### COMBINE THE MATRICES FROM 2 RUNS & ASSIGN FEATURES 1 RUN DOESN'T HAVE IN THE OTHER RUN AS "NA" VALUES  ###
 mlist<-list(counts_total_run3,counts_total_run4)
 df <- mlist[[1]]
 for (i in 2:length(mlist)) {
  df <- merge(df, mlist[[i]], by = "row.names", all=T)
  rownames(df) <- df$Row.names
 df <- df[ , !(names(df) %in% "Row.names")]  
 }

### Differential expression analysis of all biotypes included (No surrogate variable analysis done) ###
### Differential expression analysis of all biotypes included (No surrogate variable analysis done) ###
### Differential expression analysis of all biotypes included (No surrogate variable analysis done) ###
counts_total<-df
(d <- data.frame(row.names = colnames(counts_total),group=c("MSUS","Control","MSUS","Control","MSUS","Control","MSUS","Control","Control","MSUS","MSUS","Control","MSUS","Control","MSUS","Control","MSUS","Control","Control","MSUS") ))
counts_total[is.na(counts_total)] <- 0 ### the "NA" values all the matrix are substituted with 0s. 
dds <- edgeR::calcNormFactors(DGEList(counts_total), method="TMM")
dds$samples$group<-c("MSUS","Control","MSUS","Control","MSUS","Control","MSUS","Control","Control","MSUS","MSUS","Control","MSUS","Control","MSUS","Control","MSUS","Control","Control","MSUS")  ### giving the group 
dds$samples$batch<-c(1,1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2)
mm <- model.matrix(~dds$samples$batch+dds$samples$group, data=d)
mm
dds <- dds[which(filterByExpr(dds, design=mm, min.count=20, min.prop=0.9)),] ### specifying thebatch while filtering by expression 
en <- log1p(edgeR::cpm(dds)) # normalization by counts per million 



dds <- estimateDisp(dds, mm)
fit <- glmLRT(glmFit(dds, mm), coef = "dds$samples$groupMSUS")
res_without_ruvseq <- as.data.frame(topTags(fit,Inf))
head(res_without_ruvseq,n=30)
hist(res_without_ruvseq)
write_csv(res_without_ruvseq,path="/Users/alshanbayeva/Desktop/caudaEVs_story_manuscript/deepak/raw_data_excerpt/the_final_code_for_submission/caudaEVs_bothruns_allbiotypes_RUVSeq_withoutRUVSeq/res_without_ruvseq.csv")

### Note: The surrogate variable analysis was not used, since it did not affect the distribution of the samples/separation of the samples by sequencing batch ### 
```

